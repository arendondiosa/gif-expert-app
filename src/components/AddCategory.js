import React, { useState } from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import { Form, InputGroup } from "react-bootstrap";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <Form onSubmit={handleSubmit}>
      <Form.Group>
        <InputGroup>
          <Form.Control
            type="text"
            value={inputValue}
            onChange={handleInputChange}
          />
          <InputGroup.Append>
            <InputGroup.Text>
              <FontAwesomeIcon icon={faSearch} />
            </InputGroup.Text>
          </InputGroup.Append>
        </InputGroup>
        <Form.Text className="text-muted">
          Type something + Enter button
        </Form.Text>
      </Form.Group>
    </Form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
