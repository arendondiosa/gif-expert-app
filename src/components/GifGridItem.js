import React from "react";
import PropTypes from "prop-types";
import { Card } from "react-bootstrap";

export const GifGridItem = ({ title, url }) => {
  return (
    <Card>
      <Card.Img variant="top" src={url} />
      <Card.Body>
        <Card.Title>{title}</Card.Title>
      </Card.Body>
    </Card>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
