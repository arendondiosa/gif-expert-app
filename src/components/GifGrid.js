import React from "react";
import PropTypes from "prop-types";
import { CardColumns, Spinner } from "react-bootstrap";

import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);

  return (
    <div>
      <h3>{category}</h3>

      {loading && (
        <Spinner animation="border" role="status">
          <span className="sr-only">Loading...</span>
        </Spinner>
      )}

      <CardColumns>
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </CardColumns>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
