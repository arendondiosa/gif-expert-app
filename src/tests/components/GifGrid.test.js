import React from "react";
import { shallow } from "enzyme";
import { Spinner } from "react-bootstrap";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Tests <AddCategory />", () => {
  const category = "Sipmsons";

  test("Render the component", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Should display items when images are loaded using useFetchGifs", () => {
    const gifs = [
      {
        id: "one",
        url: "https://image.com/one.jpg",
        title: "Image one",
      },
      {
        id: "two",
        url: "https://image.com/two.jpg",
        title: "Image two",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find(Spinner).exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
