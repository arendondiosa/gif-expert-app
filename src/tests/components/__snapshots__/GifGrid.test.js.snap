// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests <AddCategory /> Render the component 1`] = `
<div>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Sipmsons
  </h3>
  <Spinner
    animation="border"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </Spinner>
  <CardColumns />
</div>
`;
